frontend/src/pages/team/Billing.vue

```diff
@@ -133,12 +145,31 @@ export default {
                 try {
                     billingUrl = await billingApi.getSubscriptionInfo(this.team.id)
                 } catch (err) {
-                    if (err.response.status === 404) {
-                        billingUrl = err.response.data.billingURL
+                    if (err.response.status === 402) {
+                        if (err.response.data.billingURL) {
+                            billingUrl = err.response.data.billingURL
+                        } else if (err.response.data.error) {
+                            Alerts.emit(`${this.coupon} coupon invalid`, 'warning', 7500)
+                            this.errors.coupon = `${this.coupon} is not a valid code. You will be able to provide an alternative code on the Stripe checkout page.`
+                            this.coupon = false
+                            return
+                        }
                     }
                 }
             }
             window.open(billingUrl, '_self')
+        },
+        getCookie (name) {
+            if (document.cookie) {
+                const cookies = document.cookie.split(';')
+                for (let i = 0; i < cookies.length; i++) {
+                    const cookie = cookies[i]
+                    if (cookie.split('=')[0] === name) {
```

Comment:
Given multiple cookies, the `;` is space-padded between entries so need to trim.

```suggestion
                    if (cookie.split('=')[0].trim() === name) {
```